lib/Doctrine/Migrations/Metadata/AvailableMigrationsList.php

```diff
@@ -31,16 +32,28 @@ public function getItems() : array
         return $this->items;
     }
 
-    public function getFirst(int $offset = 0) : ?AvailableMigration
+    public function getFirst(int $offset = 0) : AvailableMigration
     {
-        return $this->items[$offset] ?? null;
+        if (! isset($this->items[$offset])) {
+            throw NoMigrationsFoundWithCriteria::new('first' . ($offset>0 ? ('+' . $offset) : ''));
+        }
+
+        return $this->items[$offset];
     }
 
-    public function getLast(int $offset = 0) : ?AvailableMigration
+    public function getLast(int $offset = 0) : AvailableMigration
     {
-        return $this->items[count($this->items)-1-(-1*$offset)] ?? null;
+        $offset = count($this->items)-1-(-1*$offset);
+        if (! isset($this->items[$offset])) {
+            throw NoMigrationsFoundWithCriteria::new('first' . ($offset>0 ? ('+' . $offset) : ''));
```

Comment:
shouldn't we report it as `last -3` rather than `first +41`, as the criteria was based on the last migration ?